proto/engine/v1/electra.go

```diff
@@ -19,35 +19,45 @@ var (
 	crSize    = crExample.SizeSSZ()
 )
 
-const LenExecutionRequestsElectra = 3
+const (
+	depositRequestType       = 0
+	withdrawalRequestType    = 1
+	consolidationRequestType = 2
+)
 
 func (ebe *ExecutionBundleElectra) GetDecodedExecutionRequests() (*ExecutionRequests, error) {
 	requests := &ExecutionRequests{}
-
-	if len(ebe.ExecutionRequests) != LenExecutionRequestsElectra /* types of requests */ {
-		return nil, errors.Errorf("invalid execution request size: %d", len(ebe.ExecutionRequests))
-	}
-
-	// deposit requests
-	drs, err := unmarshalItems(ebe.ExecutionRequests[0], drSize, func() *DepositRequest { return &DepositRequest{} })
-	if err != nil {
-		return nil, err
-	}
-	requests.Deposits = drs
-
-	// withdrawal requests
-	wrs, err := unmarshalItems(ebe.ExecutionRequests[1], wrSize, func() *WithdrawalRequest { return &WithdrawalRequest{} })
-	if err != nil {
-		return nil, err
-	}
-	requests.Withdrawals = wrs
-
-	// consolidation requests
-	crs, err := unmarshalItems(ebe.ExecutionRequests[2], crSize, func() *ConsolidationRequest { return &ConsolidationRequest{} })
-	if err != nil {
-		return nil, err
+	var prevTypeNum uint8
+	for i := range ebe.ExecutionRequests {
+		requestType := ebe.ExecutionRequests[i][0]
```

Comment:
This method could panic if the request is empty slice when you index `[1:]` later. 
Consider a creating a method that handles this like:

```go
func decodeExecutionRequest(req []byte) (typ int, data []byte, err error) {
  if len(req) < 1 {
    return -1, nil, errors.New("invalid execution request, length less than 1")
  }
  return int(req[0]), req[1:], nil
}
```

Then you can use it safely in your for-range loop

```go
reqType, reqData, err := decodeExecutionRequest(ebe.ExecutionRequests[i])
```

Reply:
good catch, there's another case we should probably catch too